src/distrs/binom.jl

```diff
@@ -16,4 +16,4 @@ import .RFunctions:
 binompdf(n::Real, p::Real, k::Real) = exp(binomlogpdf(n, p, k))
 
 # logpdf for numbers with generic types
-binomlogpdf(n::Real, p::Real, k::Real) = -log1p(n) - logbeta(n - k + 1, k + 1) + k * log(p) + (n - k) * log1p(-p)
+binomlogpdf(n::Real, p::Real, k::Real) = (isinteger(k) & (zero(k) <= k <= n)) ? convert(typeof(float(p)), loggamma(n + 1) - loggamma(k + 1) - loggamma(n - k + 1) + xlogy(k, p) + xlogy(n - k, 1 - p)) : convert(typeof(float(p)), -Inf)
```

Comment:
Break this into multiple lines with a `if`. It would also be simpler/clearer to only apply the conversion once to the result, instead of repeating it.

In terms of performance, can you check the difference with the current implementation? I'm afraid it's significantly more complex, with the three calls to `loggamma`.

Reply:
> Break this into multiple lines with a if. It would also be simpler/clearer to only apply the conversion once to the result, instead of repeating it.

Will do!

As for performance, the current implementation of `logbeta` is

`logbeta(a::Number, b::Number) = loggamma(a)+loggamma(b)-loggamma(a+b)`

So if anything, the original implementation is actually less performant than the new version.

Reply:
Ok made requested changes.